docs/vault-setup-config.rst

```diff
@@ -0,0 +1,157 @@
+.. vault-setup-config:
+
+******************************
+Vault Setup and Configurations
+******************************
+
+`Vault <https://www.vaultproject.io>`_ is a tool developed by `Hashicorp <https://www.hashicorp.com>`_ that securely manages secrets - handling leasing, key revocation, key rolling, and auditing.
+
+Certificate Authorities (CAs)
+-----------------------------
+Vault is used as a Certificate Authority for Tarmak's Kubernetes Clusters.
+Three CA's are needed to provide multiple roles for each cluster as follows:
+
+* Etcd cluster that serves API server (etcd-k8s)
+    * server role for etcd daemons
+    * client role for kube-apiserver
+
+* CA for Etcd cluster that serves as overlay backend (flannel or calico) (etcd-overlay)
+    * server role for etcd daemons
+    * client role for flannel/calico daemons
+
+* Kubernetes API (k8s)
+    * master role for Kubernetes master components (kube-apiserver, kube-controller-manager, kube-scheduler)
+    * worker role for Kubernetes worker components (kube-proxy, kubelet)
+    * admin role for users requesting access to Kubernetes API as admin
+    * more specific bespoke roles that limit access e.g. read-only, namespace-specific roles
+
```

Comment:
#53 added another CA for verifying aggregated API calls. Can you include that. You can find more information here:

https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/concepts/auth.md